samza-sql-shell/src/main/java/org/apache/samza/sql/client/impl/SamzaExecutor.java

```diff
@@ -0,0 +1,582 @@
+        staticConfigs.put(SamzaSqlApplicationConfig.CFG_SQL_STMTS_JSON, JsonUtil.toJson(formatSqlStmts(statement)));
+
+        SamzaSqlApplicationRunner runner;
+        try {
+            runner = new SamzaSqlApplicationRunner(true, new MapConfig(staticConfigs));
+            runner.run();
+        } catch (SamzaException ex) {
+            m_lastErrorMsg = ex.toString();
+            LOG.error(m_lastErrorMsg);
+            return new NonQueryResult(execId, false);
+        }
+        m_executions.put(execId, runner);
+        LOG.debug("Executing sql. Id ", execId);
+
+        return new NonQueryResult(execId, true);
+    }
+
+    @Override
+    public boolean stopExecution(ExecutionContext context, int exeId) {
+        m_lastErrorMsg = "";
+
+        SamzaSqlApplicationRunner runner = m_executions.get(exeId);
+        if(runner != null) {
+            LOG.debug("Stopping execution ", exeId);
+
+            try {
+                runner.kill();
+            } catch (SamzaException ex) {
+                m_lastErrorMsg = ex.toString();
+                LOG.debug(m_lastErrorMsg);
+                return false;
+            }
+
+            try {
+                Thread.sleep(500); // wait for a second
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+
+            return true;
+        } else {
+            m_lastErrorMsg = "Trying to stop a non-existing SQL execution " + exeId;
+            LOG.warn(m_lastErrorMsg);
+            return false;
+        }
+    }
+
+    @Override
+    public boolean removeExecution(ExecutionContext context, int exeId) {
+        m_lastErrorMsg = "";
+
+        SamzaSqlApplicationRunner runner = m_executions.get(exeId);
+        if(runner != null) {
+            if (runner.status().getStatusCode().equals(ApplicationStatus.StatusCode.Running)) {
+                m_lastErrorMsg = "Trying to remove a ongoing execution " + exeId;
+                LOG.error(m_lastErrorMsg);
+                return false;
+            }
+            m_executions.remove(exeId);
+            LOG.debug("Stopping execution ", exeId);
+            return true;
+        } else {
+            m_lastErrorMsg = "Trying to remove a non-existing SQL execution " + exeId;
+            LOG.warn(m_lastErrorMsg);
+            return false;
+        }
+    }
+
+    @Override
+    public ExecutionStatus queryExecutionStatus(int execId) {
+        SamzaSqlApplicationRunner runner = m_executions.get(execId);
+        if (runner == null) {
+            return null;
+        }
+        return queryExecutionStatus(runner);
+    }
+
+    @Override
+    public String getErrorMsg() {
+        return m_lastErrorMsg;
+    }
+
+    @Override
+    public List<SqlFunction> listFunctions(ExecutionContext context) {
+        /**
+         * TODO: currently the Shell only shows some UDFs supported by Samza internally. We may need to require UDFs
```

Comment:
Can you open a Jira for this one?

Reply:
Yes, I've created a sub-task for this under jira [SAMZA-1901 ](https://issues.apache.org/jira/browse/SAMZA-1901).